service/resource/configmap/desired.go

```diff
@@ -2,8 +2,58 @@ package configmap
 
 import (
 	"context"
+
+	"github.com/prometheus/prometheus/config"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/prometheus-config-controller/service/prometheus"
 )
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	configMap, err := r.k8sClient.CoreV1().ConfigMaps(r.configMapNamespace).Get(
+		r.configMapName, metav1.GetOptions{},
+	)
+	if errors.IsNotFound(err) {
+		return nil, microerror.Maskf(configMapNotFoundError, "%s/%s", r.configMapNamespace, r.configMapName)
```

Comment:
I masked a lot of the errors here. I found it having some additional messaging useful for debugging.